Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Set default values for build arguments
+ARG DOCKERFILE_VERSION=1.0.0
+ARG AGENT_VERSION=4.3-4
+
+FROM jenkins/inbound-agent:$AGENT_VERSION-alpine
+
+USER root
+
+# Install dependencies as root
+RUN apk add --no-cache python3-dev && \
+    apk add --no-cache --virtual .build-deps libffi-dev openssl-dev gcc libc-dev make && \
+    pip3 install --no-cache-dir --upgrade pip && \
+    pip3 install --no-cache-dir docker-compose && \
+    apk del .build-deps
+
+# Note: Latest version of kubectl may be found at:
+# https://github.com/kubernetes/kubernetes/releases
+ARG KUBE_VERSION="v1.18.3"
```

Comment:
In order to use the default and allow overriding of this default you need to set the default value before the `FROM` step with the value, then after the FROM again so it can be used in the container.

So if you change it to:

```
# Set default values for build arguments
ARG DOCKERFILE_VERSION=1.0.0
ARG AGENT_VERSION=4.3-4
# Note: Latest version of kubectl may be found at:
# https://github.com/kubernetes/kubernetes/releases
ARG KUBE_VERSION=v1.18.3
# Note: Latest version of helm may be found at
# https://github.com/kubernetes/helm/releases
ARG HELM_VERSION=v3.2.3

FROM jenkins/inbound-agent:$AGENT_VERSION-alpine

....
ARG KUBE_VERSION
ARG HELM_VERSION
....
```